pkg/dryrun/dryrun_utils.go

```diff
@@ -0,0 +1,264 @@
+	for k := range inputMap {
+		keys = append(keys, k)
+	}
+
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		actualValue, exist := resultMap[key]
+		value := inputMap[key]
+
+		if !exist {
+			cmd.Println(errorColor(fmt.Sprintf("Key '%s' not found in result at path '%s'", key, parentPath)))
+
+			isStatusMatch = false
+
+			continue
+		}
+
+		path := parentPath + "." + key
+
+		switch typedV := value.(type) {
+		case string, int, int64, float64, bool:
+			v := fmt.Sprintf("%v", typedV)
+
+			av := fmt.Sprintf("%v", actualValue)
+
+			if v != av {
+				errorMatchPrint(cmd, path, v, av)
+
+				isStatusMatch = false
+			} else {
+				successMatchPrint(cmd, path, v, av)
+			}
+
+		case []interface{}:
+			// Handle array of objects and non-object arrays
+			actualArray, ok := actualValue.([]interface{})
+			if !ok {
+				notArrayPrint(cmd, path, key)
+
+				isStatusMatch = false
+
+				continue
+			}
+
+			if len(typedV) == 0 {
+				continue
+			}
+
+			switch typedV[0].(type) {
+			case map[interface{}]interface{}, map[string]interface{}:
+				allMatch := true
+				// Check all input status are match
+				for i, obj := range typedV {
+					elementMatch := false
+
+					mapObj, ok := convertMapStringKey(obj).(map[string]interface{})
+					if !ok {
+						failedParsePrint(cmd, path)
+
+						continue
+					}
+
+					for _, actualObj := range actualArray {
+						mapActualObj, ok := convertMapStringKey(actualObj).(map[string]interface{})
+						if !ok {
+							failedParsePrint(cmd, path)
+
+							continue
+						}
+
+						// If an matched element is found in the actual status, break the current loop.
+						if compareStatusObj(cmd, true,
+							fmt.Sprintf("%s[%d]", path, i), mapObj,
+							mapActualObj) {
+							elementMatch = true
+
+							break
+						}
+					}
+
+					if !elementMatch {
+						allMatch = false
+
+						errorMatchAnyArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+
+						break
+					}
+
+					successMatchArrayPrint(cmd, fmt.Sprintf("%s[%d]", path, i))
+				}
+
+				if !allMatch {
+					errorMatchArrayPrint(cmd, path)
+
+					isStatusMatch = false
+				} else {
+					successMatchArrayPrint(cmd, path)
+				}
+			default:
+				// Simple comparison of arrays eg []string
+				if reflect.DeepEqual(typedV, actualArray) {
+					successMatchPrint(cmd, path, typedV, actualArray)
+				} else {
+					errorMatchPrint(cmd, path, typedV, actualArray)
+
+					isStatusMatch = false
+				}
+			}
+
+		case map[string]interface{}:
+			if actualMap, ok := actualValue.(map[string]interface{}); ok {
+				// Recursive call for nested maps
+				isStatusMatch = compareStatusObj(cmd, isStatusMatch, path, typedV, actualMap)
+			} else {
+				failedParsePrint(cmd, path)
+
+				isStatusMatch = false
+			}
+		case interface{}:
+			typeOfTypeV := reflect.TypeOf(typedV).String()
+			typeOfActual := reflect.TypeOf(actualValue).String()
+
+			if typeOfTypeV == "map[interface {}]interface {}" &&
+				typeOfActual == "map[interface {}]interface {}" {
```

Comment:
Is this not covered by the `map[interface{}]interface{}` case above? Is there a test case that actually reaches this point?